scripts/dotnet-cli-build/PublishTargets.cs

```diff
@@ -48,12 +52,111 @@ public static BuildTargetResult Publish(BuildTargetContext c)
+                if (AzurePublisherTool.IsLatestSpecifiedVersion(targetVersionFile))
+                {
+                    AzurePublisherTool.ReleaseLeaseOnContainer(leaseId);
+                    return c.Success();
+                }
+                else
+                {
+                    // This is an old drop of latest so remove all old files to ensure a clean state
+                    AzurePublisherTool.ListBlobs($"{targetContainer}").ToList().ForEach(f => AzurePublisherTool.DeleteBlob(f));
+                }
+
+                try
+                {
+                    // Iterate over every blob in the latest version folder and copy it to the 'latest' folder
+                    foreach (string blob in _latestBlobs)
+                    {
+                        string targetUrl = $"{targetContainer}{Path.GetFileName(blob)}"
+                                        .Replace(CliNuGetVersion, "latest")
+                                        .Replace(SharedFrameworkNugetVersion, "latest");
+                        AzurePublisherTool.CopyBlob(blob.Replace("/dotnet/", ""), targetUrl);
+                    }
+
+                    // Copy the DNVM latest version files and delete the temp files
+                    IEnumerable<string> dnvmBlobs = AzurePublisherTool.ListBlobs($"{Channel}/dnvm/temp/");
+                    foreach (string blob in dnvmBlobs)
+                    {
+                        string sourceUrl = blob.Replace("/dotnet/", "");
+                        string targetUrl = $"{Channel}/dnvm/{Path.GetFileName(blob)}";
+                        AzurePublisherTool.CopyBlob(sourceUrl, targetUrl);
+                        AzurePublisherTool.DeleteBlob(sourceUrl);
+                    }
+
+                    // We finished so drop the version file signaling such for any race-condition builds (see above comment).
+                    AzurePublisherTool.DropLatestSpecifiedVersion(targetVersionFile);
+                }
+                finally
+                {
+                    AzurePublisherTool.ReleaseLeaseOnContainer(leaseId);
+                }
+            }
+
+            return c.Success();
+        }
+
+        private static bool CheckIfAllBuildsHavePublished()
+        {
+             Dictionary<string, bool> badges = new Dictionary<string, bool>()
+             {
+                 { "Windows_x86", false },
+                 { "Windows_x64", false },
+                 { "Ubuntu_x64", false },
+                 { "RHEL_x64", false },
+                 { "OSX_x64", false },
+                 { "Debian_x64", false },
+                 { "CentOS_x64", false }
+             };
+
+            _latestBlobs = new List<string>(AzurePublisherTool.ListBlobs($"{Channel}/Binaries/{CliNuGetVersion}/"));
+
+            var config = Environment.GetEnvironmentVariable("CONFIGURATION");
+            var versionBadgeName = $"{CurrentPlatform.Current}_{CurrentArchitecture.Current}";
```

Comment:
@Sridhar-MS and I thought about this before and came away blank 😄 agreed the name is weird

RuntimePlatform.Current maybe